### PR TITLE
graph conn

### LINKED ISSUE #11: Implement graph connectivity
https://github.com/theislab/scib/blob/main/scib/metrics/graph_connectivity.py

This one should use the same scipy function as scib does, and the input should be assumed to be a N by N neighbors graph

### PR DESCRIPTION
**Pull request by adamgayoso**
Fixes #11

### CODE REVIEW

File: src/scib_metrics/_graph_connectivity.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+from scipy.sparse.csgraph import connected_components
+
+
+def graph_connectivity(X: csr_matrix, labels: np.ndarray) -> float:
+    """Quantify the connectivity of the subgraph per cell type label.
+
+    Parameters
+    ----------
+    X
+        Array of shape (n_cells, n_cells) with non-zero values
```
Comment (justjhong) [inline src/scib_metrics/_graph_connectivity.py:13]: do you want to assert the k nearest neighbors?

File: src/scib_metrics/_graph_connectivity.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    for label in np.unique(labels):
+        mask = labels == label
+        graph_sub = X[mask]
+        graph_sub = graph_sub[:, mask]
+        _, comps = connected_components(graph_sub, connection="strong")
+        tab = pd.value_counts(comps)
+        clust_res.append(tab.max() / sum(tab))
+
+    return np.mean(clust_res)
```
Comment (justjhong) [inline src/scib_metrics/_graph_connectivity.py:29]: the cell types are weighted equally regardless of the proportions?

Comment (adamgayoso) [reply src/scib_metrics/_graph_connectivity.py:29]: yeah.... :(

File: src/scib_metrics/_graph_connectivity.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    Parameters
+    ----------
+    X
+        Array of shape (n_cells, n_cells) with non-zero values
+        representing distances to exactly each cell's k nearest neighbors.
+    labels
+        Array of shape (n_cells,) representing label values
+        for each cell.
+    """
+    clust_res = []
```
Comment (adamgayoso) [inline src/scib_metrics/_graph_connectivity.py:19]:
```suggestion
    """
    # TODO(adamgayoso): Utils for validating inputs
    clust_res = []
```

File: src/scib_metrics/_graph_connectivity.py (hunk 1 above)
Comment (adamgayoso) [reply src/scib_metrics/_graph_connectivity.py:13]: left as a todo as we have to make global utils for this

**Review by justjhong — APPROVED** [review]

**Pull request merged**